### DISCUSSION TITLE
Wazuh: 3rd party rule sets?

### DISCUSSION THREAD

**Discussion by wildlyunder** [Security-Onion-Solutions/securityonion#7280] [Unsupported Versions] [answered]
Hi,

Does anyone use 3rd party rule sets for Wauzh... Paid or unpaid?

If so, could you provide a link!

**Comment by dougburks** [comment, marked as answer]
While there are [multiple ruleset options for NIDS](https://docs.securityonion.net/en/2.3/rules.html#rulesets), I'm not aware of any 3rd party rule sets for Wazuh. As I mentioned at https://github.com/Security-Onion-Solutions/securityonion/discussions/7062#discussioncomment-2141197, I suspect that most organizations are simply using the default Wazuh HIDS rules and if they feel the need for additional coverage perhaps complementing with their own HIDS rules in ``local_rules.xml`` or detection logic elsewhere.

**Reply by matt00davis** [reply]
Hi

Can I just clarify

Within the native Wazuh deployment, Wazuh rules can be updated via the /var/ossec/bin/update_ruleset script which presumably updates them from here:

https://github.com/wazuh/wazuh/tree/master/ruleset/rules

Ideally the Wazuh rules could be also updated within Security Onion. If the only option is to add own rules in local_rules.xml then 'updating' existing Wazuh rules within Security Onion isn't really an option by the looks of it.

The only real options seem to be replace exiting rules with new updated rules (which could cause problems) or like you say add them to local_rules which would be a manual process. 

Is that correct or am I missing something?

Regards

Matt

**Reply by dougburks** [reply]
update_ruleset is no longer supported by Wazuh:
https://documentation.wazuh.com/current/user-manual/ruleset/update.html

As I mentioned above and at https://github.com/Security-Onion-Solutions/securityonion/discussions/7062#discussioncomment-2141197, Wazuh HIDS rules are NOT like NIDS rules where there are lots of updates every single day. If you compare Wazuh HIDS rules to NIDS rules you'll see that Wazuh HIDS rules are much more generic and abstract and that's why they don't require constant updates and additions like NIDS rules do.

**Reply by matt00davis** [reply]
Thanks Doug, understood and I get the difference between HIDS and NIDS rules. It would be good to better leverage the agent capabilities through evolving rules against Mitre TTP's for example. I get the constant updates reference but are there ever updates and if so how can they be applied?

**Reply by dougburks** [reply]
As I mentioned above and at https://github.com/Security-Onion-Solutions/securityonion/discussions/7062, it is my experience that most organizations do not update the default Wazuh rules. They may add their own rules to local_rules.xml as described at https://docs.securityonion.net/en/2.3/wazuh.html#tuning-rules or detection logic elsewhere in Security Onion.

Please keep in mind that signature-based detection like HIDS, NIDS, and antivirus will catch some but not all attacks. Complementing the default rules with your own custom signature-based detection for your environment and the adversaries that you face will catch more attacks, but still not all. That's why threat hunting is so vitally important.

**Reply by matt00davis** [reply]
Thanks for your time Doug, much appreciated

**Reply by Mastadamus** [reply]
I utilize a non default local_rules.xml ruleset that is mostly based of sigma rules and sysmon. It works pretty well.

**Comment by wildlyunder** [comment]
We've gone down the route of working out how to pull the rules ourselves and adding them. We don't want to be in a situation of finding out there is a Zero Day; having to pull the rule manually then applying it. We want to automate to reduce the time of indication.
